openwisp_notifications/handlers.py

```diff
@@ -203,25 +203,61 @@ def send_email_notification(sender, instance, created, **kwargs):
     else:
         target_url = None
     if target_url:
-        description += _('\n\nFor more information see %(target_url)s.') % {
-            'target_url': target_url
-        }
-
-    send_email(
-        subject,
-        description,
-        instance.message,
-        recipients=[instance.recipient.email],
-        extra_context={
-            'call_to_action_url': target_url,
-            'call_to_action_text': _('Find out more'),
-        },
-    )
+        description += _('\n\nFor more information see %(target_url)s.') % {'target_url': target_url}
 
-    # flag as emailed
-    instance.emailed = True
-    # bulk_update is used to prevent emitting post_save signal
-    Notification.objects.bulk_update([instance], ['emailed'])
+    return True, subject, description, target_url
+
+
+@receiver(post_save, sender=Notification, dispatch_uid='send_email_notification')
+def send_email_notification(sender, instance, created, **kwargs):
+    # Abort if a new notification is not created
+    if not created:
+        return
+
+    can_send, subject, description, target_url = can_send_email(instance)
```

Comment:
Why did we require to refactor this logic?

I am not convinced by the return signature of the `can_send_email` function. E.g. if the notification is malformed, `can_send_email` will only return `False` (1 return value). I believe the tuple unpacking would fail in this scenario.